src/network/MissionControlTasks.cpp

```diff
@@ -127,28 +127,37 @@ CameraStreamTask::CameraStreamTask(websocket::SingleClientWSServer& server)
 	: util::AsyncTask<>("MCP_Stream"), _server(server) {}
 
 void CameraStreamTask::openStream(const CameraID& cam, int fps) {
-	std::lock_guard lock(_mutex);
-	_open_streams[cam] = 0;
-	auto it = Constants::video::STREAM_RFS.find(cam);
-	int rf = it != Constants::video::STREAM_RFS.end() ? it->second
-													  : Constants::video::H264_RF_CONSTANT;
-	_camera_encoders[cam] = std::make_shared<video::H264Encoder>(fps, rf);
+	// run async since opening a camera can take a while
+	if (_open_streams.find(cam) == _open_streams.end()) {
+		std::thread([this, cam, fps]() {
+			std::lock_guard lock(_mutex);
+			auto it = Constants::video::STREAM_RFS.find(cam);
+			int rf = it != Constants::video::STREAM_RFS.end()
+						 ? it->second
+						 : Constants::video::H264_RF_CONSTANT;
+			auto enc = std::make_shared<video::H264Encoder>(fps, rf);
+			auto cam_handle = robot::openCamera(cam);
+			if (cam_handle) {
+				_open_streams.insert_or_assign(cam, stream_data_t(enc, cam_handle));
+			}
+		}).detach();
```

Comment:
The jetson can only open 2 cameras at a time (due to usb bandwidth issues? not sure) so this code opens cameras if and when we need them.

Reference counting makes sure that when we drop all references to an open camera it is automatically closed.